lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -48,6 +64,35 @@ def outdated_instances
 
     private
 
+    def execute_ssh(ssh_debug: false)
+      Retriable.retriable(base_interval: 5, tries: 3) do
+        @command_response = run_ssh_command(
+          first_ec2_instance, '/bin/true', ssh_debug: ssh_debug
+        )
+        raise StandardError.new, 'Validation failed.' unless
+          @command_response.exitstatus.zero?
+        @ilog.info(@command_response.output.split(/\r\n/)) if ssh_debug
+        @command_response.exitstatus.zero?
+      end
+    end
+
+    def run_ssh_command(instance, command, timeout_connection: true,
+                        ssh_debug: false)
+
+      remote_cmd =
+        build_command(instance, command, timeout_connection, ssh_debug)
+      Moonshot::SSHForkExecutor.new.run(remote_cmd)
+    end
+
+    def build_command(instance, command, timeout_connection, ssh_debug)
+      remote_cmd = "#{@ssh_user}@#{instance.public_dns_name}"
+      command_options = 'ssh '
+      options = ['UserKnownHostsFile=/dev/null', 'StrictHostKeyChecking=no']
+      command_options += '-v ' if ssh_debug
+      command_options += '-o ConnectTimeout=10' if timeout_connection
+      "#{command_options} -o #{options.join(' -o ')} #{remote_cmd} #{command}"
+    end
+
```

Comment:
All this ssh code here is to run a specific command `/bin/true` which seems a test command which makes me feel that this code does not belong to this ASG class, but to other specific to execute tests/checks

Reply:
Moving it to a new module `ASGChecks`